core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -98,13 +98,26 @@ public Builder type(String type) {
         }
 
         @Override
+        public Builder fieldDataSettings(Settings fieldDataSettings) {
+            this.fieldDataSettings = fieldDataSettings;
+            return this;
+        }
+
+        @Override
         public ParentFieldMapper build(BuilderContext context) {
             if (type == null) {
                 throw new MapperParsingException("[_parent] field mapping must contain the [type] option");
             }
-            setupFieldType(context);
-            fieldType.setHasDocValues(context.indexCreatedVersion().onOrAfter(Version.V_2_0_0_beta1));
-            return new ParentFieldMapper(fieldType, type, context.indexSettings());
+            MappedFieldType joinFieldType = Defaults.JOIN_FIELD_TYPE.clone();
```

Comment:
Why not keep this as a member of the builder, and have setting field data settings set the fielddatatype on it directly?